test/proxy/utils/UUPSUpgradeable.test.js

```diff
@@ -66,7 +68,27 @@ contract('UUPSUpgradeable', function (accounts) {
     // infinite loop reverts when a nested call is out-of-gas
     await expectRevert(
       this.instance.upgradeTo(otherInstance.address),
-      'Address: low-level delegate call failed',
+      'Function must not be called through delegatecall',
     );
   });
+
+  describe('compatibility with legacy version of UUPSUpgradeable', function () {
+    for (const artefact of Legacy) {
+      it(`can upgrade from ${artefact._json.contractName}`, async function () {
+        const legacyImpl = await artefact.new();
+        const legacyInstance = await ERC1967Proxy.new(legacyImpl.address, '0x')
+          .then(({ address }) => artefact.at(address));
+
+        const receipt = await legacyInstance.upgradeTo(this.implInitial.address);
+
+        const UpgradedEvents = receipt.logs.filter(({ address, event }) =>
+          address === legacyInstance.address &&
+          event === 'Upgraded',
+        );
+        expect(UpgradedEvents.length).to.be.equal(1);
+
+        expectEvent(receipt, 'Upgraded', { implementation: this.implInitial.address });
+      });
```

Comment:
Let's add an additional assertion to check the upgrade did in fact go through, interacting with the proxy and seeing it delegates to the new implementation.